### ISSUE TITLE
Select text between closest bookmarks

### ISSUE THREAD

**Issue by jung-kurt** [haqk/micro-bookmark#5]
Selecting a large number of lines with the keyboard can be unwieldy in a terminal editor such as Micro. One solution would be to use your first-rate plugin to help with the task. One scheme would be to introduce two new bookmark commands: one that selects the text from the current line to the closest preceding bookmark, and another that selects to the closest following bookmark. However, a single command that selects text between the closest bookmarks might be conceptually easier to use and remember. One would then just move the cursor away from a bookmarked line before issuing the command.

Is this something that you would accept as a pull request?

**Comment by haqk** [comment]
Text selection is beyond the scope for this plugin.

However, you may find this useful:  [How can I mark select text?](https://github.com/zyedidia/micro/issues/2990)

**Comment by haqk** [comment]
Reopening — I'd closed this as out of scope, but on reflection it's a natural fit, and micro now exposes everything needed (`Cursor:SetSelectionStart`/`SetSelectionEnd` plus the `buffer.Loc(x, y)` Lua constructor), so I've implemented it.

Shipped in **v2.5.0** as three commands. All operate on the active list, select whole lines, and are unbound by default:

- `selectBetweenBookmarks` — select the lines between the two bookmarks bracketing the cursor (place the cursor between two marks; it reports if you're on a mark or outside the marked range)
- `selectToNextBookmark` — select from the current line down to the nearest bookmark below
- `selectToPrevBookmark` — select from the nearest bookmark above up to the current line

Run them from the command bar (`Ctrl-E`, then the command name) or bind them in `bindings.json`. Both your suggested designs are covered: the directional pair, and the single "between" command.

Available now from the repo (and `repo.json`); it'll reach `micro -plugin install bookmark` once the plugin-channel update lands. Thanks for the suggestion.
